clientconn.go

```diff
@@ -118,6 +118,15 @@ func WithPicker(p Picker) DialOption {
 // WithBackoffConfig configures the dialer to use the provided backoff
 // parameters after connection failures.
 func WithBackoffConfig(b *BackoffConfig) DialOption {
```

Comment:
I see WithMaxBackoffDelay could be helpful and used by most of users who want to tune this behavior. I do see it adds an additional exported function. I am fine with having WithBackoffConfig only. But I feel the following code is probably better:

func WithBackoffConfig(b *BackoffConfig) DialOption {
d := DefaultBackoffConfig
if b.MaxDelay > 0 {
d.MaxDealy = b.MaxDelay
}
return withbackoff(&d)
}

And please describe this behavior somehow in the comments.